src/kademlia/node.cpp

```diff
@@ -966,8 +966,12 @@ void node::incoming_request(msg const& m, entry& e)
 		};
 
 		// attempt to parse the message
+		// also reject the message if it has any non-fatal encoding errors
+		// because put messages contain a signed value they must have correct bencoding
+		// otherwise re-encoding the value will break the signature
```

Comment:
yeah, perhaps it does make sense to keep the main decode path as fast as possible.